torchchat/generate.py

```diff
@@ -786,24 +790,41 @@ def _gen_model_input(
         assert (
             max_new_tokens is not None
         ), "max_new_tokens must be specified for Flamingo models"
-        assert isinstance(
-            prompt, str
-        ), "(Currently) prompt must be a str for Flamingo models"
-
-        is_multimodal = images is not None
-        content = [{"type": "text", "content": prompt}]
-
-        if is_multimodal:
-            content = [{"type": "image", "content": images[0]}] + content
 
-        messages = [
-            Message(
-                role="user",
-                content=content,
-                eot=True,
-            ),
-            Message(role="assistant", content=""),
-        ]
+        image_found = False
+        messages = []
+        for message in prompt:
```

Comment:
This would be `torchchat.py generate Llama3.2-11B` right?

Since it sends `prompt: str` and uses the image_prompt field

You might need to "create" a container prompt with those 2

Reply:
Or chat calls a curried version of this function that creates the format before calling this function